core/src/main/java/io/grpc/internal/StatsTraceContext.java

```diff
@@ -233,14 +249,21 @@ void callEnded(Status status) {
       uncompressedBytesSentMetric = RpcConstants.RPC_SERVER_UNCOMPRESSED_RESPONSE_BYTES;
       uncompressedBytesReceivedMetric = RpcConstants.RPC_SERVER_UNCOMPRESSED_REQUEST_BYTES;
     }
-    statsCtx
-        .with(RpcConstants.RPC_STATUS, TagValue.create(status.getCode().toString()))
-        .record(MeasurementMap.builder()
-            .put(latencyMetric, stopwatch.elapsed(TimeUnit.MILLISECONDS))
-            .put(wireBytesSentMetric, wireBytesSent)
-            .put(wireBytesReceivedMetric, wireBytesReceived)
-            .put(uncompressedBytesSentMetric, uncompressedBytesSent)
-            .put(uncompressedBytesReceivedMetric, uncompressedBytesReceived)
-            .build());
+    long roundtripNanos = stopwatch.elapsed(TimeUnit.NANOSECONDS);
+    MeasurementMap.Builder builder = MeasurementMap.builder()
+        .put(latencyMetric, roundtripNanos / NANOS_PER_MILLI)
```

Comment:
No, the measurements are doubles. Although the unit is millisecond, it can still retain nano-level precision.